### DISCUSSION TITLE
WebReader 'rotate' function

### DISCUSSION THREAD

**Discussion by Mangleuz** [gotson/komga#1568] [Ideas]
### Describe your suggested feature

Folks with laptops do not want to make the entire OS to rotate, in order just to rotate the pages in the cbr/cbz/pdf to fit the screen nicely. Normally 'Rotate' from menu (or 'r') would be the way to go. Such feature would really add something important making komga preferable to other alternatives.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the latest version.
- [X] I will fill out all of the requested information in this form.

**Comment by gotson** [comment]
Why not rotate the source file? I don't understand what the problem is.

**Comment by Mangleuz** [comment]
> Why not rotate the source file? I don't understand what the problem is.

Thank you for reply. When I read your reply I first thought I was being trolled, but then I saw that you are the repository owner 😄 

Take a look at these two screenshots from my browser using Calibre-Web: 
90 degrees [portrait](https://i.ibb.co/PTnhSqk/Capture1.png) vs [landscape](https://i.ibb.co/CMqctH6/Capture2.png).

If i were to rotate my 50 GB's of source files it would besides the hours of processing take 100 GB's instead and also be doubling the amount of files and folders. 
Its much better to instead press 'r' like in Calibre-Web. But Komga is in every regard much nicer than Calibre-Web if it was not for the lack of rotation.

**Comment by gotson** [comment]
i still don't understand what you are trying to achieve, Is it because the pages are in portrait, and your screen is in landscape ?

In that case, how does that work once you've rotated in the client ? You physically put your laptop on the side so that the screen is at 90 degree angle?

**Comment by Mangleuz** [comment]
> You physically put your laptop on the side so that the screen is at 90 degree angle?

Thank you for your patience and efforts to understand me. 

Yes, exactly. Most of the ppl I know with laptops do precisely this; holding the laptop like an ordinary book making maximum use of the size of the entire screen. Each page fits nicely that way, with no need of scrolling making the reading more immersive. (Lol, me and gf occasionally turn our massive 43-inch TV screen 90° to read massive hi-res graphical novels 😄). I'll happily ditch Calibre-Web if i can do the same with Komga.

**Comment by mondays0902** [comment]
This feature is really useful. The setup of (additional) vertical screen is quite common for reading nowaday: social media(facebook, insta, etc.), webpage, books, etc due to advantage of content/size matching.

<img width="650" height="488" alt="image" src="https://github.com/user-attachments/assets/2a3dbd46-afbd-4653-acec-9731db7dfd9f" />

I believe that main benefiter of this feature will be desktop users like me. At this moment, to achieve this I must remote access my PC and use any app support rotating, instead of using Komga web reader. Considering mobile users (still like me because I'm using both), it's not really urgent. But I think mobile users will use app like Tachiyomi or Paperback for better reading features( like cache locally). So in my point of view, itis really helpful for me to have this feature

**Reply by TiggsPanther** [reply]
Very much this! 
Especially in the, admittedly rare but not nonexistant, case of slightly meta comics. Where some panels (or even just speech bubbles) on the same page are rotated differently.

Not a problem on iPad, somewhat difficult when reading on desktop
